### PR TITLE
[Ready][?] Robustin's cult nerf, back from the dead.

### PR DESCRIPTION
**Pull request by lilacrhododendron**
:cl: Robustin and Subject217
balance: Once the blood cult reaches 20% of the active player population, they will receive a notice that the cult is "rising" - and after a moderate delay the eyes of all existing and new cultists will be permanently red. Examining a cultist with uncovered eyes will confirm their supernatural appearance.
balance: Once the blood cult reaches 40% of the active player population they will receive a warning and after a moderate delay will glow red, permanently revealing the identity of existing and new blood cultists.
add: Any non-cultist can now strike a (filled) soulshard with a bible to purify it. Purified shades have a unique appearance and will be eager to get revenge on the cult. Bibles can be printed in the library.
balance: Juggernauts have 25% less HP and 35% less projectile reflection.
balance: Cult mirror shield is slightly easier to break, has less illusions, and has -33% throwing distance.
balance: The EMP blood spell has -1 light and heavy radius.
balance: The revive rune now requires 3 sacrifices per revive, it still starts with one "freebie" revive. Giving "souls" to AFK/Catatonic cultists remains free of charge.
balance: Twisted Construction now has a channel time with added noise/effects when used on doors. It also hurts you slightly more to use.
balance: You can now only hold 1 blood spell without an empowering rune and 4 with one.
del: The Bloody Bastard Sword is no longer available in Blood Cult. It's still spawnable with admin tools.
/:cl:

I really liked these changes, and I was sad to see it go unmerged. I played this version of cult several times during test-merges and it fixed a lot of the issues. I've put most of Robustin's changes in here.

As far as balance is concerned in retrospect given the small changes in the past few months, I reviewed them all, and I don't believe that the changes in this PR need any modification. Willing to continue tweaking Cult as necessary.

On the Bastard Sword: It's a really weird unintuitive item that a lot of people complain about. Naksu told me to remove cult's ability to make it, so that's what I've done. Cult in general is filled to the brim with bloat items that do something slightly differently without being very valuable so it's not a great loss.

### CODE REVIEW

**Comment by Wilchenx** [comment]
image

oof

**Comment by subject217** [comment]
The only check causing trouble is MapDiffBot. This PR contains zero map edits, so something's fucky with it. It has nothing to do with the PR itself.

File: code/__DEFINES/cult.dm
Comment (ExcessiveUseOfCobblestone) [inline code/__DEFINES/cult.dm]: 15 and/or 16 need a comment stating what they do

File: code/game/gamemodes/cult/cult.dm
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/cult/cult.dm]: undocumented?

File: code/game/objects/items/storage/book.dm
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/storage/book.dm]: target should be user (target is for location of progress bar only)

File: code/game/objects/items/storage/book.dm
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/items/storage/book.dm]: soulstones will only ever have a shade in them outside of foolish mememins, so make it typeless.

Surprised soulstones just don't have a trapped_mob var or something

File: code/modules/antagonists/cult/blood_magic.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm]: undocumented?

File: code/modules/antagonists/cult/blood_magic.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm]: undocumented?

File: code/modules/antagonists/cult/blood_magic.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/blood_magic.dm]: target user

File: code/modules/antagonists/cult/cult.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/cult.dm]: suggestion: 1 proc for both rise/ascend with a second argument (there is not ever a time you will ascend without rising, but you will always rise before ascending)

File: code/modules/antagonists/cult/cult_items.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/cult_items.dm]: undocumented?

File: code/modules/antagonists/cult/runes.dm
Comment (ExcessiveUseOfCobblestone) [inline code/modules/antagonists/cult/runes.dm]: you can just delete this, looks like debug hooplah

that or move it to... click *shudders*

File: code/modules/antagonists/cult/cult.dm
Comment (ExcessiveUseOfCobblestone) [reply code/modules/antagonists/cult/cult.dm]: you could actually replace the proc with `try_to_rise(mob/living/cultist)` and put the checks in there, so the copypaste of `if(cult_team.cult_risen)` blah blah blah is not necessary.

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
Some (albeit minor) undocumented changes :(

I didn't review the tooltip stuff cuz that is ^^ here and I am ___ here

**Comment by subject217** [comment]
looks like robustin was tinkering with shit and left the PR in an unfinished state, that explains why it wasn't merged :^)

i'll push some fixes when i get some spare time

**Comment by subject217** [comment]
In my experience it's only really good if you get a bunch of people trapped in a small area. Otherwise, you can just kite, and the mirror shield with the regular cult sword is better. With the deconversion changes that happened in previous PRs and this PR massively nerfing mass conversion and making several smaller nerfs to various Cult combat mechanics, it's probably not an issue anymore. Assuming I can fix this and get it merged.

File: code/game/gamemodes/cult/cult.dm
Comment (vuonojenmustaturska) [reply code/game/gamemodes/cult/cult.dm]: this was IIRC a fix for a situation where the math worked in a funny way to prevent the gamemode from launching

File: code/game/objects/items/storage/book.dm
Comment (vuonojenmustaturska) [reply code/game/objects/items/storage/book.dm]: disagree, this is fine

**Comment by HydroFloric** [comment]
Holy shit finally, maybe once this gets merged nu cult wont be as shit.

**Comment by Tlaltecuhtli** [comment]
can you add an examinate desc effect when someone is deconverted that says he got deconverted and it stays for like 2 min? sec can barely catch cultists and usually will holy water 3-4 times the same person because they arent sure he got deconverted or not

File: code/game/objects/items/storage/book.dm
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/storage/book.dm]: reeee when would a soulstone ever have anything outside of a shade >:(

**Comment by ExcessiveUseOfCobblestone** [comment]
@Tlaltecuhtli with this they shouldn't have the outline/eyes anymore.

**Comment by BeeSting12** [comment]
The two people that downvoted probably dont play otherwise they'd know how stupid powerful cult is.

**Comment by vuonojenmustaturska** [comment]
As I said earlier,  feel free to cut the tooltip stuff

**Comment by subject217** [comment]
oh boy

at this point it's probably less work for me to revert and redo the diffs with the dumb fluff cut out and the bastard sword removal

File: code/game/gamemodes/cult/cult.dm
Comment (subject217) [reply code/game/gamemodes/cult/cult.dm]: I can remove it or I can add a changelog, won't put it in the CL though unless I understand what exactly it's there for.

File: code/modules/antagonists/cult/blood_magic.dm
Comment (subject217) [reply code/modules/antagonists/cult/blood_magic.dm]: fixed in changelog

**Comment by JJRcop** [comment]
I can squash everything for you if you want

**Comment by subject217** [comment]
I know how to squash commits, this isn't web-edited. At first I was just doing it that way because it was lazy, now I'm doing it because I got burned on squashing commits and had to manually edit diffs out instead of just reverting a commit. So I'm doing it this way.

File: code/modules/antagonists/cult/cult.dm
Comment (subject217) [reply code/modules/antagonists/cult/cult.dm]: you're welcome to code this but if a maintainer doesn't have a problem with it i'd really rather leave it as is

**Comment by subject217** [comment]
Tentatively ready. I really can't test a lot of this shit locally, so a test merge would be nice.

File: code/modules/antagonists/cult/cult.dm
Comment (ExcessiveUseOfCobblestone) [reply code/modules/antagonists/cult/cult.dm]: only problem with it currently is that there's the same copypasta thrown around so if someone wanted to add a tertiary item, or simply edit the current checks, they'd have to sift through multiple areas which gives avenues to bugs.

I don't mind free GBP tho :^)

File: code/__DEFINES/cult.dm
Comment (ExcessiveUseOfCobblestone) [reply code/__DEFINES/cult.dm]: `Percent before Rise/Ascend` would be sufficient.

File: code/modules/antagonists/cult/cult.dm
Comment (AnturK) [inline code/modules/antagonists/cult/cult.dm]: Technically alive can be zero here.

File: code/modules/mob/living/carbon/human/examine.dm
Comment (AnturK) [inline code/modules/mob/living/carbon/human/examine.dm]: This is kinda leaky if we allow pure red as eye color

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/examine.dm
Comment (subject217) [reply code/modules/mob/living/carbon/human/examine.dm]: That's what the examine text is for, no? It makes sure there's no doubt that they're a cultist.

Comment (ChangelingRain) [reply code/modules/mob/living/carbon/human/examine.dm]: Pure red is allowed, so people who become cultists with naturally pure red eyes will have this desc.

Comment (subject217) [reply code/modules/mob/living/carbon/human/examine.dm]: I understand the issue. It seems pretty unlikely to happen unless you were explicitly trying to make it happen. I'm not sure how to fix this. I suppose I could check to see if the cult was already risen, or maybe it's possible to make BLOODCULT_EYE something that isn't selectable as an eye color, but the more issues with this PR the more I feel like I'm in over my head with code that I only partially understand.

File: code/modules/antagonists/cult/cult.dm
Comment (subject217) [reply code/modules/antagonists/cult/cult.dm]: This compiles, but doesn't that divide by zero? Or does BYOND just not care if it's 0/0?

Comment (AnturK) [reply code/modules/antagonists/cult/cult.dm]: Not sure what you mean but this is not compile time value, so byond has no idea if this is division by zero.

File: code/modules/mob/living/carbon/human/examine.dm
Comment (AnturK) [reply code/modules/mob/living/carbon/human/examine.dm]: You can just add simple mob trait for this.

Comment (ChangelingRain) [reply code/modules/mob/living/carbon/human/examine.dm]: Yeeeah you underestimate how many people have pure red eyes(the answer is too many)

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/examine.dm]: I think my sybil character does actually

File: code/modules/antagonists/cult/cult.dm
Comment (subject217) [reply code/modules/antagonists/cult/cult.dm]: This one I can't figure out. This is for removing the halo and the eyes upon deconversion, right? As far as I can tell the halo overlay isn't tied to the head. I looked around a little bit and I figured out how to cut a specific overlay, but I haven't found any way to just update the head icons. Or maybe I'm misunderstanding.

**Comment by subject217** [comment]
attempting to bait a review/update on my questions...........

File: code/modules/antagonists/cult/cult.dm
Comment (AnturK) [reply code/modules/antagonists/cult/cult.dm]: Just needs `update_body` call instead.

File: code/modules/mob/living/carbon/human/examine.dm
Comment (AnturK) [reply code/modules/mob/living/carbon/human/examine.dm]: Up to you, might be nice for admin abuse if it doesn't check specific color.

File: code/modules/antagonists/cult/cult.dm
Comment (AnturK) [reply code/modules/antagonists/cult/cult.dm]: If there's no cultyplayers or alive you can just return since it will won't need update then.

**Comment by Mey-Ha-Zah** [comment]
Yeah I'm gonna say, the spell nerfs and the removal of the bastard sword is okay, but the fucking hell sign over every head is bullshit, revs don't get that kinda of "OH TARGET ME". It just serves to make targets for the crew, the red eyes I can stand, maybe force them to only speak Nar Nar after a certain percent, but a HUGE FUCKING Emblem marking them as a target is the worst possible thing. It makes any misdirection impossible.

**Pull request merged**
